src/charm.py

```diff
@@ -156,16 +156,10 @@ def _make_ingress_config(self) -> list:
         site = config.get("site")
 
         relation = None
-        try:
-            relation = self.model.get_relation("ingress-proxy")
-            # XXX: better handle when service-hostname isn't available via
-            # replace due to racing with other charm rather than wrap in a
-            # try/except KeyError.
-            if relation:
-                backend_site_name = relation.data[relation.app]["service-hostname"]
-                site = backend_site_name
-        except KeyError:
-            pass
+        relation = self.model.get_relation("ingress-proxy")
+        if relation:
+            backend_site_name = relation.data[relation.app]["service-hostname"]
+            site = backend_site_name
```

Comment:
Why not assigning directly `site = relation.data[relation.app]["service-hostname"]` and removing the second assignment?

Reply:
What I meant is that you don't need the variable `backend_site_name` at all. It's only used to reassign its value to `site`